### PR TITLE
Add support for custom column names and YAML format output to fix ave/time

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This adds YAML format output to fix ave/time and also allows to customize the column name using the same approach
as used with `thermo_style yaml` and `dump yaml`. Also more detailed examples for how to read/process and plot data are added to the structured data Howto.

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

YAML format thermo output and scalar output from fix ave/time use the exact same data structures (only that the fix will only create a single YAML "document").

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

### CODE REVIEW

File: doc/src/fix_ave_time.rst [hunk 1]
```diff
@@ -258,10 +258,16 @@ each input value specified in the fix ave/time command.  For *mode* =
 scalar, this means a single line is written each time output is
 performed.  Thus the file ends up to be a series of lines, i.e. one
 column of numbers for each input value.  For *mode* = vector, an array
-of numbers is written each time output is performed.  The number of
-rows is the length of the input vectors, and the number of columns is
-the number of values.  Thus the file ends up to be a series of these
-array sections.
+of numbers is written each time output is performed.  The number of rows
+is the length of the input vectors, and the number of columns is the
+number of values.  Thus the file ends up to be a series of these array
```
Comment (sjplimp) [inline doc/src/fix_ave_time.rst:263]: Pargraph break at "If the filename ..."

File: doc/src/fix_ave_time.rst [hunk 2]
```diff
@@ -258,10 +258,16 @@ each input value specified in the fix ave/time command.  For *mode* =
...
+sections.  If the filename ends in '.yaml' or '.yml' then the output
+format is conforming to the `YAML standard <https://yaml.org/>`_ which
```
Comment (sjplimp) [inline doc/src/fix_ave_time.rst:265]: is conforming --> conforms to

File: doc/src/fix_ave_time.rst [hunk 3]
```diff
@@ -258,10 +258,16 @@ each input value specified in the fix ave/time command.  For *mode* =
...
+allows to easily import that data into tools and scripts that support
```
Comment (sjplimp) [inline doc/src/fix_ave_time.rst:266]: allows easy import of the data into

File: doc/src/fix_ave_time.rst [hunk 4]
```diff
@@ -307,8 +313,10 @@ appropriate fields from the fix ave/time command.
 Restart, fix_modify, output, run start/stop, minimize info
 """""""""""""""""""""""""""""""""""""""""""""""""""""""""""
 
-No information about this fix is written to :doc:`binary restart files <restart>`.  None of the :doc:`fix_modify <fix_modify>` options
-are relevant to this fix.
+No information about this fix is written to :doc:`binary restart files
+<restart>`.  The :doc:`fix_modify colname <fix_modify>` options can be
```
Comment (sjplimp) [inline doc/src/fix_ave_time.rst:317]: option, not options.  Colname is one opttion

File: doc/src/fix_modify.rst [hunk 5]
```diff
@@ -12,7 +12,7 @@ Syntax
 
 * fix-ID = ID of the fix to modify
 * one or more keyword/value pairs may be appended
-* keyword = *temp* or *press* or *energy* or *virial* or *respa* or *dynamic/dof* or *bodyforces*
+* keyword = *temp* or *press* or *energy* or *virial* or *respa* or *dynamic/dof* or *bodyforces* or *colname*
```
Comment (sjplimp) [inline doc/src/fix_modify.rst:15]: these should probably be alphebetized, now that there are so many

Comment (akohlmey) [reply doc/src/fix_modify.rst]: done

**Review by sjplimp — APPROVED** [review]

**Pull request merged**